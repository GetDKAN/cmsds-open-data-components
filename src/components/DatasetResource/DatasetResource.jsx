import React from 'react';
import PropTypes from 'prop-types';
import DataTableToolbar from '../DataTableToolbar/DataTableToolbar';
import DataTable from '../dataset/DataTable/DataTable';

const DatasetResource = ({
  resource,
  datasetTitle,
  datasetDescription,
  datasetModified,
  datasetReleased,
  datasetRefresh,
}) => {
  return (
    <div
      id={`resource_${resource.identifier}`}
      className="resource-table"
      data-testid="data-table"
      data-resource-identifier={resource.identifier}
    >
      {/* Filter logic here */}

      <DataTableToolbar
        datasetTitle={datasetTitle}
        datasetDescription={datasetDescription}
        datasetModified={datasetModified}
        datasetReleased={datasetReleased}
        datasetRefresh={datasetRefresh}
        instanceId={1}
      />
      <DataTable datasetTitle={datasetTitle} />
    </div>
  );
};

DatasetResource.propTypes = {
  /**
   * Dataset data object
   */
  resource: PropTypes.object,
  /**
   * Dataset title
   */
  datasetTitle: PropTypes.string,
  /**
   * Dataset description
   */
  datasetDescription: PropTypes.string,
  /**
   * Dataset modified date string
   */
  datasetModified: PropTypes.string,
  /**
   * Dataset released date string
   */
  datasetReleased: PropTypes.string,
  /**
   * Dataset anticipated refresh date string
   */
  datasetRefresh: PropTypes.string,
};

DatasetResource.displayName = 'DatasetResource';
export default DatasetResource;
