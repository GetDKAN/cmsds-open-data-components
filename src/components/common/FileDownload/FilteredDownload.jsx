import React from 'react'
import PropTypes from 'prop-types'
import FontAwesomePro from '../FontAwesomePro/FontAwesomePro'
import DatasetDownloadLink from '../../dataset/DatasetDownloadLink/DatasetDownloadLink'

const FilteredDownload = ({ resource, datasetTitle, filteredURL, isDatasetExplorerDownloadLink = false }) => {
  let format = null

  if (
    Object.prototype.hasOwnProperty.call(resource, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data, 'format')
  ) {
    format = resource.data.format
    format.toUpperCase()
  }

  let title = null

  if (
    Object.prototype.hasOwnProperty.call(resource, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data.data, 'title')
  ) {
    title = resource.data.data.title
  } else {
    title = 'This Dataset'
  }

  const item = (
    <>
      {filteredURL && title && format && (
        <DatasetDownloadLink
          linkText={(
            <>
              {isDatasetExplorerDownloadLink ? (
                <FontAwesomePro icon='download' aria-hidden />
              ) : (
                <FontAwesomePro icon='filter' aria-hidden />
              )}
              {isDatasetExplorerDownloadLink ? (
                <span>Download filtered dataset ({format.toUpperCase()})</span>
              ) : (
                <span>Download filtered dataset <span className='sr-only'>- {format}</span></span>
              )}
            </>
          )}
          linkHref={filteredURL}
        />
      )}
      {!isDatasetExplorerDownloadLink && (
        <div className='format'>
          <span>
            {format}
          </span>
        </div>
      )}
    </>
  )

  return isDatasetExplorerDownloadLink ? item : <div className='resource'>{item}</div>
}

FilteredDownload.propTypes = {
  /**
   * Dataset data object
   */
  resource: PropTypes.shape({
    data: PropTypes.object,
    format: PropTypes.string,
    identifier: PropTypes.string
  }),
  /**
   * Dataset title
   */
  datasetTitle: PropTypes.string,
  /**
   * Filtered dataset download url
   */
  filteredURL: PropTypes.string,
  /**
   * Dataset explorer download link gets different styling
   */
  isDatasetExplorerDownloadLink: PropTypes.bool
}
FilteredDownload.displayName = 'FilteredDownload'
export default FilteredDownload
