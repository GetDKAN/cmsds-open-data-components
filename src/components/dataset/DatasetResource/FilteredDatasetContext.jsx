import { createContext } from 'react'

export const FilteredDispatch = createContext(null)

export const defaultResourceState = {
  columnOrder: [],
  columns: [],
  count: 0,
  unfilteredCount: 0,
  unfilteredColumnCount: 0,
  currentPage: 0,
  density: 'density-3',
  excludedColumns: {},
  filters: [],
  loading: false,
  limit: 20,
  sort: [],
  values: [],
  downloadSize: {
    sizeLabel: 'KB',
    size: 0
  }
}
