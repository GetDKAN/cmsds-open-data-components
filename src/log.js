import isString from 'lodash/isString'
import find from 'lodash/find'
import toUpper from 'lodash/toUpper'
import toLower from 'lodash/toLower'
import findIndex from 'lodash/findIndex'
import config from './config/index'
import { isIeUserAgent } from './utilities/displayUtilities'

let { logLevel } = config

const userAgent = navigator.userAgent || ''
const isIE = isIeUserAgent(userAgent)

// force IE 11 to log errors only to prevent crashing the browser
logLevel = isIE ? 'ERROR' : toUpper(logLevel)

const levels = [
  {
    key: 'DEBUG',
    logFunc: console.log
  },
  {
    key: 'INFO',
    logFunc: console.info
  },
  {
    key: 'WARN',
    logFunc: console.warn
  },
  {
    key: 'ERROR',
    logFunc: console.error
  }
]

const logLevelObject = find(levels, { key: logLevel })
const logLevelIndex = logLevelObject
  ? findIndex(levels, { key: logLevel })
  : false
const allowedLevelKeys =
  logLevelIndex === false
    ? []
    : levels.slice(logLevelIndex, levels.length).map(({ key }) => key)

const log = ({ key, logFunc }) => (args) => {
  const prefix = `[${key}]`
  if (isString(args[0])) {
    args[0] = `${prefix} ${args[0]}`
  } else {
    args.push(prefix)
  }

  logFunc.apply(null, args)
}

const logger = {}

levels.forEach((level) => {
  const key = level.key
  logger[toLower(key)] = (...args) => {
    if (allowedLevelKeys.indexOf(key) > -1) {
      log(level)(args)
    }
  }
})

export default logger
