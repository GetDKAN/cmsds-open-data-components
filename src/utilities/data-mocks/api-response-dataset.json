{
    "fixture": true,
    "accessLevel": "public",
    "landingPage": "https://data.cms.gov/provider-data/dataset/test-data-abcd-0005",
    "bureauCode": [
        "009:38"
    ],
    "issued": "2020-01-20",
    "@type": "dcat:Dataset",
    "modified": "2020-01-20",
    "released": "2020-09-17",
    "archiveExclude": false,
    "keyword": [
        {
            "identifier": "8a0cf0a8-a3d0-5223-b1cc-82fbb3101cdb",
            "data": "hhc"
        },
        {
            "identifier": "9929edf6-e8af-5e74-b28e-6c43c0e0a534",
            "data": "general information"
        }
    ],
    "contactPoint": {
        "@type": "vcard:Contact",
        "fn": "Home Health Compare",
        "hasURL": "https://www.cms.gov/apps/contacts"
    },
    "publisher": {
        "identifier": "4c1de1c4-f083-5877-8968-7ddc04756c49",
        "data": {
            "@type": "org:Organization",
            "name": "Centers for Medicare & Medicaid Services (CMS)"
        }
    },
    "identifier": "test-data-abcd-0005",
    "description": "Home Health Care - Measure Date Range",
    "title": "Home Health Care - Measure Date Range",
    "programCode": [
        "009:000"
    ],
    "distribution": [
        {
            "identifier": "716256f0-6d17-56a6-9b5d-c35879011e9e",
            "data": {
                "@type": "dcat:Distribution",
                "downloadURL": "https://pqdc-dkan.ddev.site/provider-data/sites/default/files/resources/2850c4af459ff447d4e9bada0637ec54_1701355540/Test_Data_abcd_0005.csv",
                "mediaType": "text/csv",
                "%Ref:downloadURL": [
                    {
                        "identifier": "2850c4af459ff447d4e9bada0637ec54__1701355540__source",
                        "data": {
                            "filePath": "s3://913461122956-pdc-dev-test-minimal-data/Test_Data_abcd_0005.csv",
                            "identifier": "2850c4af459ff447d4e9bada0637ec54",
                            "mimeType": "text/csv",
                            "perspective": "source",
                            "version": "1701355540",
                            "checksum": null
                        }
                    },
                    {
                        "identifier": "2850c4af459ff447d4e9bada0637ec54__1701355540__local_url",
                        "data": {
                            "filePath": "https://h-o.st/provider-data/sites/default/files/resources/2850c4af459ff447d4e9bada0637ec54_1701355540/Test_Data_abcd_0005.csv",
                            "identifier": "2850c4af459ff447d4e9bada0637ec54",
                            "mimeType": "text/csv",
                            "perspective": "local_url",
                            "version": "1701355540",
                            "checksum": null
                        }
                    }
                ]
            },
            "format": "csv"
        }
    ],
    "theme": [
        {
            "identifier": "eda174d2-96df-5537-868f-f5ed625c3ede",
            "data": "Home health services"
        }
    ],
    "%modified": "2023-11-30T09:45:33-0500"
}
