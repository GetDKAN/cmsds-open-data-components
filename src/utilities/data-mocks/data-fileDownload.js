export const resourceData = {
  data: {
    identifier: '11745687-dc47-52ac-b9e4-66c50fe772ad',
    data: {
      '@type': 'dcat:Distribution',
      title: 'Test Data abcd 0016',
      downloadURL: 'https://pqdc-dkan.ddev.site/provider-data/sites/default/files/resources/f69b073f217ea35ed14997178f7085c8_1677006237/Test_Data_abcd_0016.csv',
      mediaType: 'text/csv',
      '%Ref:downloadURL': [
        {
          identifier: 'f69b073f217ea35ed14997178f7085c8__1677006237__source',
          data: {
            filePath: 's3://913461122956-pdc-dev-test-minimal-data/Test_Data_abcd_0016.csv',
            identifier: 'f69b073f217ea35ed14997178f7085c8',
            mimeType: 'text/csv',
            perspective: 'source',
            version: '1677006237',
            checksum: null
          }
        },
        {
          identifier: 'f69b073f217ea35ed14997178f7085c8__1677006237__local_url',
          data: {
            filePath: 'http://h-o.st/provider-data/sites/default/files/resources/f69b073f217ea35ed14997178f7085c8_1677006237/Test_Data_abcd_0016.csv',
            identifier: 'f69b073f217ea35ed14997178f7085c8',
            mimeType: 'text/csv',
            perspective: 'local_url',
            version: '1677006237',
            checksum: null
          }
        }
      ]
    },
    format: 'csv'
  }
}

export const responseHeaders = {
  'content-length': 99999
}
