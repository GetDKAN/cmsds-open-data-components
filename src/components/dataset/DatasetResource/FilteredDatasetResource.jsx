import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { FilteredDispatch } from './FilteredDatasetContext'
import useDatastore from '../../../hooks/useDataStore'
import { getApiBaseUrl } from '../../../utilities/getApiBaseUrl'

import {
  useTable,
  usePagination,
  useFilters,
  useSortBy,
  useFlexLayout,
  useResizeColumns,
  useColumnOrder
} from 'react-table'

const FilteredDatasetResource = ({
  children,
  resource
}) => {
  const [filters, setFilters] = useState([{ column: '', condition: '', value: '' }]) // keep track of the applied filters
  const [filtersApplied, setFiltersApplied] = useState([])
  const [initCol, setInitCol] = useState([]) // we want to save a reference to filter against
  const [col, setCol] = useState([]) // columns we pass to table, post filter
  const [data, setData] = useState([]) // these are our cell values
  const [initOrder, setInitOrder] = useState([]) // initial order saved as a reference
  const [curOrder, setCurOrder] = useState([]) // current order that we pass to the data table
  const [visCol, setVisCol] = useState([]) // our visible column reference to prevent more datastore queries than need be
  const [activeDensity, setActiveDensity] = useState('Normal')
  const [downloadSize, setDownloadSize] = useState({
    sizeLabel: 'KB',
    size: 0
  })
  const prepareColumns = (columns, schema) => {
    return columns.map((column) => ({
      Header: schema && schema.fields[column].description ? schema.fields[column].description : column,
      accessor: column
    }))
  }
  const baseUrl = getApiBaseUrl()
  const filteredResource = useDatastore(
    '',
    baseUrl,
    {
      limit: 20,
      manual: true
    },
    {}
    // additionalParams
  )

  // listen to our filter resource changing, and flip the go switch once we have the info to query.
  useEffect(() => {
    if (resource.identifier) {
      filteredResource.setResource(resource.identifier)
      filteredResource.setManual(false)
    }
    if (!resource.identifier) {
      setInitOrder([])
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [resource])

  useEffect(() => {
    if (visCol.length) {
      const newCol = initCol.filter((item) => visCol.includes(item.accessor))
      setCol(newCol)
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [visCol])

  // the issue is that this fires more than expected.
  useEffect(() => {
    if (Object.keys(filteredResource.schema).length > 0) {
      setData(filteredResource.values)
      // we only want to initialize columns if we don't have any info
      if (!initCol.length) {
        const newColumns = prepareColumns(filteredResource.columns, filteredResource.schema[Object.keys(filteredResource.schema)])
        setCol(newColumns) // init
        setInitCol(newColumns) // init
      }
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [filteredResource.loading])

  const filterTypes = React.useMemo(
    () => ({
      // Add a new fuzzyTextFilterFn filter type.
      // fuzzyText: fuzzyTextFilterFn,
      // Or, override the default text filter to use
      // "startWith"
      text: (rows, id, filterValue) => (
        rows.filter((row) => {
          const rowValue = row.values[id]
          return rowValue !== undefined
            ? String(rowValue)
              .toLowerCase()
              .startsWith(String(filterValue).toLowerCase())
            : true
        })
      )
    }),
    []
  )

  const defaultColumn = React.useMemo(
    () => ({
      // Let's set up our default Filter UI
      // Filter: DefaultColumnFilter,
      Filter: false,
      minWidth: 30,
      // width: 150,
      maxWidth: 400
    }),
    []
  )
  const filteredTable = useTable(
    {
      columns: col,
      data,
      initialState: { pageIndex: Number(filteredResource.offset / filteredResource.limit), pageSize: 20 },
      manualPagination: true,
      manualSortBy: true,
      manualFilters: true,
      pageCount: Math.ceil(Number(filteredResource.count) / filteredResource.limit),
      defaultColumn,
      filterTypes
      // NextPage and previous Page are defined methods in the Dataset explorer instead
    },
    useFilters,
    useFlexLayout,
    useResizeColumns,
    useColumnOrder,
    useSortBy,
    usePagination
  )

  // Initialize current and init orders
  useEffect(() => {
    if (filteredTable.allColumns.length && !initOrder.length) {
      setInitOrder(filteredTable.allColumns)
      setCurOrder(filteredTable.allColumns.map((d) => d.id))
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [filteredTable.allColumns.length, initOrder.length])

  const resetVisibility = () => {
    const initialCol = initCol.map((item) => item.accessor)
    setVisCol(initialCol)
  }

  return (
    <FilteredDispatch.Provider
      value={{
        filteredResource,
        filteredTable,
        initOrder,
        curOrder,
        setCurOrder,
        setVisCol,
        resetVisibility,
        initCol,
        visCol,
        activeDensity,
        setActiveDensity,
        filters,
        setFilters,
        filtersApplied,
        setFiltersApplied,
        downloadSize,
        setDownloadSize
      }}
    >
      {children}
    </FilteredDispatch.Provider>
  )
}
FilteredDatasetResource.propTypes = {
  children: PropTypes.node,
  identifier: PropTypes.string,
  resource: PropTypes.object
}
FilteredDatasetResource.displayName = 'FilteredDatasetResource'
export default FilteredDatasetResource
