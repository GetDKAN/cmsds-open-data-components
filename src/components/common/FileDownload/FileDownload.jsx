import React, { useContext, useEffect } from 'react';
import PropTypes from 'prop-types';
import FontAwesomePro from '../FontAwesomePro/FontAwesomePro';
import { formatBytes } from '../../../utilities/downloadUtilities';
import { pdcAxiosInstance } from '../../../api/axiosInstance';
import log from '../../../log';
import DatasetDownloadLink from '../../dataset/DatasetDownloadLink/DatasetDownloadLink';
import { getApiBaseUrl } from '../../../utilities/getApiBaseUrl';
import { FilteredDispatch } from '../../DatasetResource/FilteredDatasetContext';

function FileDownload(props) {
  const { resource, isDatasetExplorerDownloadLink = false } = props;
  let downloadURL = null;

  const { downloadSize, setDownloadSize } = useContext(FilteredDispatch);

  if (
    Object.prototype.hasOwnProperty.call(resource, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data.data, 'downloadURL')
  ) {
    const fixedURL = getApiBaseUrl(false);
    downloadURL = `${fixedURL}${resource.data.data.downloadURL.split('/provider-data')[1]}`;
  }

  let format = null;

  if (
    Object.prototype.hasOwnProperty.call(resource, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data, 'format')
  ) {
    format = resource.data.format;
    format.toUpperCase();
  }

  let title = null;
  if (
    Object.prototype.hasOwnProperty.call(resource, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data, 'data') &&
    Object.prototype.hasOwnProperty.call(resource.data.data, 'title')
  ) {
    title = resource.data.data.title;
  } else {
    title = 'This Dataset';
  }

  useEffect(() => {
    if (downloadURL) {
      pdcAxiosInstance
        .head(downloadURL)
        .then(({ headers }) => {
          const cLength = headers['content-length'];
          setDownloadSize(formatBytes(cLength));
        })
        .catch((e) => {
          log.debug('Error retrieving fileSize, FileDownload: ', e);
        });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [downloadURL]);

  const item = (
    <>
      {downloadURL && title && format && (
        <DatasetDownloadLink
          linkText={
            <>
              <FontAwesomePro icon="download" aria-hidden />
              {isDatasetExplorerDownloadLink ? (
                <span>
                  Download full dataset ({format.toUpperCase()}){' '}
                  {`${downloadSize.size} ${downloadSize.sizeLabel}`}
                </span>
              ) : (
                <span>
                  Download full dataset{' '}
                  <span className="sr-only">
                    - {format} - {`${downloadSize.size} ${downloadSize.sizeLabel}`}
                  </span>
                </span>
              )}
            </>
          }
          linkHref={downloadURL}
        />
      )}
      {!isDatasetExplorerDownloadLink && (
        <div className="format">
          <span className="format-inner">{format}</span>
          {downloadSize && (
            <>
              <span className="bullet-point">&bull;</span>
              <span>{`${downloadSize.size} ${downloadSize.sizeLabel}`}</span>
            </>
          )}
        </div>
      )}
    </>
  );

  return isDatasetExplorerDownloadLink ? item : <div className="resource">{item}</div>;
}

FileDownload.propTypes = {
  /**
   * Dataset data object
   */
  resource: PropTypes.shape({
    data: PropTypes.object,
    format: PropTypes.string,
    identifier: PropTypes.string,
  }),
  /**
   * Dataset explorer download link gets different styling
   */
  isDatasetExplorerDownloadLink: PropTypes.bool,
};
FileDownload.displayName = 'FileDownload';
export default FileDownload;
