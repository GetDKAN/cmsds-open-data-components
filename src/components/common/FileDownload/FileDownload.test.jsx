import { render, screen } from '@testing-library/react'
import FileDownload from './FileDownload'
import { act } from '@testing-library/react'
import { resourceData, responseHeaders } from '../../../utilities/data-mocks/data-fileDownload'
import axiosMockAdapter from '../../../stories/utilities/axios-mock'
import log from '../../../log'
import { FilteredDispatch } from '../../dataset/DatasetResource/FilteredDatasetContext'
import { MemoryRouter } from 'react-router-dom'

// Mock the config file
vi.mock('../../../../config', () => ({
  default: {
    site: "https://pqdc-dkan.ddev.site/",
    domainsNeedProviderDataPath: []
  }
}))

// Mock window.location
global.window = Object.create(window)
Object.defineProperty(window, 'location', {
  value: {
     protocol: 'http:',
     hostname: 'localhost'
  },
  writable: true
})

const componentArgs = {
  resource: resourceData,
  isDatasetExplorerDownloadLink: false
}


const renderComponent = (args = componentArgs) => (
  render(
    <MemoryRouter
      initialEntries={[{
        pathname: '/',
        search: '',
        hash: '',
        state: {
          fromTitle: '',
          fromUrl: ''
        },
        key: 'abc123'
      }]}
    >
      <FilteredDispatch.Provider
        value={{
          downloadSize: {
            size: 10,
            sizeLabel: 'KB'
          }
        }}
      >
        <FileDownload {...args} />
      </FilteredDispatch.Provider>
    </MemoryRouter>
  )
)

describe('FileDownload component.', () => {
  it('Matches snapshot.', () => {
    const renderedFileDownload = renderComponent()

    expect(renderedFileDownload.asFragment()).toMatchSnapshot()
  })

  it('Does not render download link if resource prop data is not provided.', () => {
    renderComponent({
      resource: {}
    })

    expect(screen.queryByText('Download full dataset')).not.toBeInTheDocument()
  })

  it('File download size displays.', async () => {
    // Mock api request inside FileDownload
    axiosMockAdapter.onHead(resourceData.data.downloadURL).reply(200, '', responseHeaders)

    window.location.protocol = 'https:',
    window.location.hostname = 'pqdc-dkan.ddev.site'

    await act(async () => {
      renderComponent()
    })

    expect(screen.queryByText('10 KB')).toBeInTheDocument()
  })

  it('Dataset explorer download link is formatted correctly.', async () => {
    renderComponent({
      ...componentArgs,
      isDatasetExplorerDownloadLink: true
    })

    expect(screen.queryByText('Download full dataset (CSV) 10 KB')).toBeInTheDocument()
  })

  it('Error is caught when api request fails.', async () => {
    // Mock failed api request inside FileDownload
    axiosMockAdapter.onHead(resourceData.data.downloadURL).reply(500, '', {}, {})
    const logDebug = vi.spyOn(log, 'debug')

    await act(async () => {
      renderComponent()
    })

    expect(logDebug).toHaveBeenCalledWith('Error retrieving fileSize, FileDownload: ', expect.objectContaining({}))
  })
})
