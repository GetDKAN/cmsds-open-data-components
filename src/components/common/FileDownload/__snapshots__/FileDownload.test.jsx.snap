// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FileDownload component. > Matches snapshot. 1`] = `
<DocumentFragment>
  <div
    class="resource"
  >
    <a
      class="dataset-download-link"
      href=""
      rel="noopener noreferrer"
      role="button"
      tabindex="0"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        aria-label="download"
        class="fa fa-download"
        focusable="false"
        height="16"
        role="img"
        viewBox="0 0 576 512"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M528 288h-92.1l46.1-46.1c30.1-30.1 8.8-81.9-33.9-81.9h-64V48c0-26.5-21.5-48-48-48h-96c-26.5 0-48 21.5-48 48v112h-64c-42.6 0-64.2 51.7-33.9 81.9l46.1 46.1H48c-26.5 0-48 21.5-48 48v128c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V336c0-26.5-21.5-48-48-48zm-400-80h112V48h96v160h112L288 368 128 208zm400 256H48V336h140.1l65.9 65.9c18.8 18.8 49.1 18.7 67.9 0l65.9-65.9H528v128zm-88-64c0-13.3 10.7-24 24-24s24 10.7 24 24-10.7 24-24 24-24-10.7-24-24z"
          fill="#323A45"
        />
      </svg>
      <span>
        Download full dataset 
        <span
          class="sr-only"
        >
          - csv - 10 KB
        </span>
      </span>
    </a>
    <div
      class="ds-c-dialog-wrap"
    >
      <dialog
        aria-labelledby="dialog--1__heading"
        class="ds-c-dialog csv-download-link-dialog"
        id="dialog--1"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="ds-c-dialog__window"
        >
          <div
            class="ds-c-dialog__header"
          >
            <h2
              class="ds-c-dialog__heading"
              id="dialog--1__heading"
            >
              Notice: Downloading in Excel
            </h2>
            <button
              aria-label="Close notice dialog"
              class="ds-c-close-button ds-c-dialog__close"
              id="dialog--1__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="ds-c-icon ds-c-icon--close ds-c-icon--close-thin "
                id="dialog--1__close__icon"
                viewBox="-2 -2 18 18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 13.0332964L13.0332964 0M13.0332964 13.0332964L0 0"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
          <div
            class="ds-c-dialog__body"
          >
            <div
              id="dialog--1__content"
            >
              <p>
                Before downloading these files in Excel, please review the 
                <strong>
                  Excel Download Instructions
                </strong>
                 to ensure accuracy with the display of your data when downloading, opening and saving these files.
              </p>
              <ul
                class="csv-faq-link"
              >
                <li>
                  <a
                    href="/about#download-files-in-excel"
                  >
                    Excel Download Instructions
                  </a>
                </li>
              </ul>
              <p>
                Would you like to proceed?
              </p>
            </div>
            <div
              class="ds-c-dialog__actions"
            >
              <div
                class="csv-download-link-dialog-actions"
              >
                <button
                  class="btn blueberry"
                  data-download-url="https://pqdc-dkan.ddev.site/sites/default/files/resources/f69b073f217ea35ed14997178f7085c8_1677006237/Test_Data_abcd_0016.csv"
                >
                  Yes, download
                </button>
                <button
                  class="btn"
                >
                  No, cancel
                </button>
              </div>
            </div>
          </div>
        </div>
      </dialog>
    </div>
    <div
      class="format"
    >
      <span
        class="format-inner"
      >
        csv
      </span>
      <span
        class="bullet-point"
      >
        â€¢
      </span>
      <span>
        10 KB
      </span>
    </div>
  </div>
</DocumentFragment>
`;
